domino/domino.py

```diff
@@ -58,17 +65,25 @@ def _configure_logging(self):
         logging.basicConfig(level=logging_level)
         self._logger = logging.getLogger(__name__)
 
-    def _initialise_request_manager(self, api_key, domino_token_file):
-        if api_key is None and domino_token_file is None:
-            raise Exception("Either api_key or path_to_domino_token_file "
-                            f"must be provided via class constructor or "
-                            f"environment variable")
-        elif domino_token_file is not None:
-            self._logger.info("Initializing python-domino with bearer token auth")
-            return _HttpRequestManager(BearerAuth(domino_token_file))
-        else:
-            self._logger.info("Fallback: Initializing python-domino with basic auth")
-            return _HttpRequestManager(HTTPBasicAuth('', api_key))
+    def _authenticate(self, api_key=None, auth_token=None, domino_token_file=None):
+        """
+        Method to authenticate the request manager. An existing domino client object can use
+        this with a new token if the existing credentials expire.
+
+        If no auth is provided, the call fails with:
+            AssertionError: Unable to authenticate: no authentication value provided in
+                            {'api_key': None, 'auth_token': None, 'domino_token_file': None}
+        """
+        auth_types = {"api_key": api_key,
+                      "auth_token": auth_token,
+                      "domino_token_file": domino_token_file}
+
+        assert any(auth_type for auth_type in auth_types.values()), \
+                  (f"Unable to authenticate: no authentication value provided in "
+                   f"{pprint.pformat(auth_types)}")
+
+        self.request_manager = _HttpRequestManager(get_auth_by_type(**auth_types))
+        return self.request_manager
```

Comment:
Reading again and coming back to this, I can see the reason for needing `_authenticate()`. However, I think it can be made to be very thin. Happy to talk offline about options if necessary.

Reply:
> I think it can be made to be very thin.

Done. Thanks for the help.